recipes/hazelcast-cpp-client/4.0.0/conanfile.py

```diff
@@ -0,0 +1,86 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class HazelcastCxx(ConanFile):
+    name = "hazelcast-cpp-client"
+    description = "C++ client library for Hazelcast in-memory database."
+    license = "Apache-2.0"
+    topics = ("conan", "hazelcast", "client", "database", "cache")
+    homepage = "https://github.com/hazelcast/hazelcast-cpp-client"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.os == "Linux" and self.settings.compiler.libcxx!="libstdc++11":
+            raise ConanInvalidConfiguration("Requires settings.compiler.libcxx = libstdc++11")
```

Comment:
Is it really impossible to build with libc++ or the old libstdc++?

Reply:
Yes, the library requires C++11 support as the min. requirement.

Reply:
It works the other way.
`libc++` works perfectly fine with C++11 / 14 / 17 / 20. So it should work here, if the project is written without some assumptions of gcc's implementation of stdlib.
AFAIK, even `libstdc++` should work. It's about ABI, which was broken by C++11 for strings in GCC many years ago.

So, I don't like this check.

Reply:
I submitted a commit to see the affect of removing this check. https://github.com/conan-io/conan-center-index/pull/4745/commits/af6fb2e2b272f6028c0019de0da4ea7494212640 

AFAIK, it was causing some checks to fail.

Reply:
It seems to work just fine, which is great 👍